playbooks/roles/setup_pipeline/tasks/main.yml

```diff
@@ -0,0 +1,137 @@
+---
+- name: create application user
+  user:
+    name: "{{ pipeline_user }}"
+    home: "{{ pipeline_app_dir }}"
+    createhome: no
+    shell: /bin/bash
+    generate_ssh_key: yes
+  tags:
+    - install
+    - install:base
+
+- name: create edxapp user dirs
+  file:
+    path: "{{ item.path }}"
+    state: directory
+    owner: "{{ pipeline_user }}"
+    group: "{{ common_web_group }}"
+    mode: "{{ item.mode | default(0755) }}"
+  with_items:
+    - { path: "{{ pipeline_app_dir }}" }
+    # needed for the ansible 1.5 git module
+    - { path: "{{ pipeline_app_dir }}/.ssh" }
+    - { path: "{{ pipeline_hadoop_tasks_dir }}" }
+    - { path: "{{ pipeline_venv_dir }}" }
+  tags:
+    - install
+    - install:base
+
+- name: own key authorized
+  file:
+    src: "{{ pipeline_app_dir }}/.ssh/id_rsa.pub"
+    dest: "{{ pipeline_app_dir }}/.ssh/authorized_keys"
+    owner: "{{ pipeline_user }}"
+    group: "{{ pipeline_group }}"
+    force: yes
+    state: link
+
+- name: ensure user is in sudoers
+  lineinfile: >
+    dest=/etc/sudoers state=present
+    regexp='^%pipeline ALL\=' line='%pipeline ALL=(ALL) NOPASSWD:ALL'
+    validate='visudo -cf %s'
+
+- name: create edxapp log dir
+  file:
+    path: "{{ edxapp_log_dir }}"
+    state: directory
+    owner: "{{ common_log_user }}"
+    group: "{{ common_log_user }}"
+  tags:
+    - install
+    - install:base
+
+# Do A Checkout
+- name: checkout edx-analytics-pipeline repo into {{ pipeline_code_dir }}
+  git_2_0_1:
+    dest: "{{ pipeline_code_dir }}"
+    repo: "{{ pipeline_repo }}"
+    version: "{{ pipeline_version }}"
+    accept_hostkey: yes
+  become_user: "{{ pipeline_user }}"
+  #environment:
+  #  GIT_SSH: "{{ edxapp_git_ssh }}"
+  register: pipeline_checkout
+  tags:
+    - install
+    - install:code
+
+- name: git clean after checking out edx-platform
+  shell: cd {{ pipeline_code_dir }} && git clean -xdf
+  become_user: "{{ pipeline_user }}"
+  tags:
+    - install
+    - install:code
+
+- name: Create the virtualenv to install the Python requirements
+  command: >
```

Comment:
FWIW, here's an example I used for a Django demo site I'm working on for Django Boston:

```
# Create virtualenv
- name: Create djsite virtualenv and packages
  pip:
    requirements: "{{ djsite_app_dir }}requirements.txt"
    virtualenv: "/home/djsite/venvs/djsite"
    virtualenv_python: python3.5
```

It is not really important but for reference, `djsite_app_dir` is a full path to a subdir in the djsite user home.

I also specified the python version. This is only supported in Ansible 2.0+

For quick ref: http://docs.ansible.com/ansible/pip_module.html